pyuvdata/uvdata/uvh5.py

```diff
@@ -189,256 +194,619 @@ def _get_compression(compression):
+        associated, AND the blt ordering to either be (time, baseline) or
+        (baseline,time).
         """
-        Read header information from a UVH5 file.
+        if self.__blts_are_rectangular is not None:
+            return self.__blts_are_rectangular
+
+        if self.Nblts == self.Ntimes * self.Nbls and self.blt_order in (
+            ("time", "baseline"),
+            ("baseline", "time"),
+        ):
+            return True
+
+        is_rect, self.__time_first = uvutils.determine_rectangularity(
+            time_array=self.time_array,
+            baseline_array=self.baseline_array,
+            nbls=self.Nbls,
+            ntimes=self.Ntimes,
+        )
+        return is_rect
+
+    @cached_property
+    def _time_first(self) -> bool:
+        """Whether times move first in the blt axis."""
+        # first hit the blts_are_rectangular property to set the time_first property
+        if not self.blts_are_rectangular:
+            return False
+        if self.__time_first is not None:
+            return self.__time_first
+        if self.Ntimes == 1:
+            return False
+        if self.Nbls == 1:
+            return True
+        return self.header["time_array"][1] != self.header["time_array"][0]
+
+    @cached_property
+    def phase_center_catalog(self) -> dict | None:
+        """Dictionary of phase centers."""
+        header = self.header
+        if "phase_center_catalog" not in header:
+            return None
+        phase_center_catalog = {}
+        key_list = list(header["phase_center_catalog"].keys())
+        if isinstance(header["phase_center_catalog"][key_list[0]], h5py.Group):
+            # This is the new, correct way
+            for pc, pc_dict in header["phase_center_catalog"].items():
+                pc_id = int(pc)
+                phase_center_catalog[pc_id] = {}
+                for key, dset in pc_dict.items():
+                    if issubclass(dset.dtype.type, np.bytes_):
+                        phase_center_catalog[pc_id][key] = bytes(dset[()]).decode(
+                            "utf8"
+                        )
+                    elif dset.shape is None:
+                        phase_center_catalog[pc_id][key] = None
+                    else:
+                        phase_center_catalog[pc_id][key] = dset[()]
+        else:
+            # This is the old way this was written
+            for key in header["phase_center_catalog"].keys():
+                pc_dict = json.loads(
+                    bytes(header["phase_center_catalog"][key][()]).decode("utf8")
+                )
+                pc_dict["cat_name"] = key
+                pc_id = pc_dict.pop("cat_id")
+                phase_center_catalog[pc_id] = pc_dict
+
+        return phase_center_catalog
+
+    @cached_property
+    def phase_type(self) -> str:
+        """The phase type of the data."""
+        h = self.header
+        if "phase_type" not in h:
+            return "drift"
```

Comment:
I'm not sure this is what we want. In new "phase_type" will not be present (it's being replaced by the phase center catalog) but that doesn't mean it's a drift type.

I'd suggest checking if phase_center_catalog is present and if it is, this should return "drift" only if all the "cat_type"s are "unprojected", otherwise return "phased".

Reply:
Thank you, I was really struggling with the logic implied by the earlier code.

Reply:
Can you ensure the logic now is correct?

Reply:
It looks correct to me now.